src/index.ts

```diff
@@ -81,12 +87,38 @@ const supportedLocalesSearch = supportedLocales.length > 0
   ? `&supportedLocales=${supportedLocales.join(',')}`
   : ''
 
-const sessionPromise = fetchWithRetry(`${rootPath}/api/sessions${window.location.search}${items}${supportedLocalesSearch}`, {
-  body: '{}',
-  credentials: 'same-origin',
-  headers: new Headers({ 'Content-Type': 'application/json' }),
-  method: 'POST',
-}).catch(err => console.log('Error while loading session with error: ', err));
+const bindingIdSearch = bindingId
+  ? `&__bindingId=${bindingId}`
+  : ''
+
+const createInitialSessionRequest = () => {
+  return fetchWithRetry(`${rootPath}/api/sessions${window.location.search}${items}${supportedLocalesSearch}${bindingIdSearch}`, {
+    body: '{}',
+    credentials: 'same-origin',
+    headers: new Headers({ 'Content-Type': 'application/json' }),
+    method: 'POST',
+  })
+}
+
+const clearSession = () => {
+  return fetchWithRetry(`${rootPath}/api/sessions/invalidToken?items=*`, {
+    credentials: 'same-origin',
+    headers: new Headers({ 'Content-Type': 'application/json' }),
+    method: 'GET',
+  }, 1)
+}
+
+const onError = (err: any) => console.log('Error while loading session with error: ', err)
+
+let sessionPromise: Promise<void | SessionResponse>
+if (bindingChanged) {
+  sessionPromise = clearSession()
+    .then(createInitialSessionRequest)
```

Comment:
Just to be sure, this will make `sessionPromise` a promise, and not a function right? The same as ` sessionPromise = createInitialSessionRequest()`